puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy

```diff
@@ -15,16 +15,26 @@ pipeline {
             steps {
 
                 deleteDir()
-                ghprb_git_checkout()
-
+                if (package_action == 'scratch') {
+                    ghprb_git_checkout()
+                }
+                else if (package_action == 'release') {
+                    git url: 'https://github.com/theforeman/foreman-packaging', branch: "${env.headCommit}"
+                }
             }
         }
 
         stage('Find Packages to Build') {
             steps {
 
                 script {
-                    changed_packages = sh(returnStdout: true, script: "git diff origin/${ghprbTargetBranch} --name-only -- 'packages/**.spec'").trim()
+                    def changed_packages
+                    if (package_action == 'scratch') {
+                        changed_packages = sh(returnStdout: true, script: "git diff origin/${ghprbTargetBranch} --name-only -- 'packages/**.spec'").trim()
+                    }
+                    else if (package_action == 'release') {
+                        changed_packages = sh(returnStdout: true, script: "git diff ${env.beforeCommit}...${env.headCommit} --name-only -- 'packages/**.spec'").trim()
```

Comment:
Looks like a re-usable function now that just needs passing in the thing to be diffed

Reply:
Those env variables set by the git or github plugin?

Reply:
no, those are randomly chosen params, that have to be filled in by a greater god, erm, `prprocessor` as the plugins don't know what we are building :(